include/wabt/ir.h

```diff
@@ -259,8 +284,8 @@ struct FuncSignature {
   // So to use this type we need to translate its name into
   // a proper index from the module type section.
   // This is the mapping from parameter/result index to its name.
-  std::unordered_map<uint32_t, std::string> param_type_names;
-  std::unordered_map<uint32_t, std::string> result_type_names;
+  std::unordered_map<uint32_t, Var> param_type_names;
```

Comment:
If the name is not found, we need to throw an error. A string has no location. I think a Var is nicer than creating a name / location pair.